### DISCUSSION TITLE
AutoRotate result

### DISCUSSION THREAD

**Discussion by MichelMichels** [micjahn/ZXing.Net#462] [Q&A]
Hi, is it possible to see if a detected barcode was `AutoRotate`d?

I have a case where a CODE 128 barcode is detected, but the resultpoints I get back are incompatible with the original image and I suppose that I have to rotate the resultpoints to get the original coordinates on the image.

**Comment by MichelMichels** [comment]
I created a repository to showcase my problem: https://github.com/MichelMichels/zxing-code-128-bug

The code reads an image file (through SkiaSharp) and draws a red rectangle around the barcode when found. Also draws the value of the barcode above the barcode.

**Comment by MichelMichels** [comment]
I fixed it in a hacky way:

```csharp
var barcodeWidth = Math.Abs(lastPoint.X - firstPoint.X);
    var barcodeHeight = 100f;

    SKPoint leftPoint, rightPoint;
    if (firstPoint.X > lastPoint.X)
    {
        // image is rotated 180 degrees
        leftPoint = new SKPoint(outputImage.Width - firstPoint.X, outputImage.Height - barcodeHeight - firstPoint.Y );
        rightPoint = new SKPoint(outputImage.Width - lastPoint.X, outputImage.Height - barcodeHeight - lastPoint.Y);
    } else
    {
        leftPoint = firstPoint;
        rightPoint = lastPoint;
    }
```

**Comment by micjahn** [comment]
The property `result.ResultMetadata[ResultMetadataType.ORIENTATION]` contains the rotation in degrees (multiples of 90 degrees).
